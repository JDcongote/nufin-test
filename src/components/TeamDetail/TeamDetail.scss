@import '../../styles/variables/index';
@import '../../styles/tools/mixins';

.team-detail {
  width: 100vw;
  background-color: var(--color--background-primary);
  position: fixed;
  top: var(--header--height);
  &__title {
    justify-content: center;
    align-content: center;
    display: flex;
    font-family: var(--font--college);
    font-size: 40px;
    font-weight: 100;
    position: relative;
    margin: 1rem 0;
    @include breakpoint(phone-up) {
      font-size: 60px;
    }
    .title {
      &__highlight {
        opacity: 0.4;
        width: 100%;
        height: 100%;
        position: absolute;
      }
    }
  }
  &__player-card {
    display: flex;
    height: 140px;
    max-height: 140px;
    background: var(--color--secondary);
    margin: 0 1em;
    border-radius: var(--border--radius--small);
    font-weight: bold;
    -webkit-box-shadow: 0px 0px 5px 0px rgba(0, 0, 0, 0.18);
    -moz-box-shadow: 0px 0px 5px 0px rgba(0, 0, 0, 0.18);
    box-shadow: 0px 0px 5px 0px rgba(0, 0, 0, 0.18);
    .player-card {
      &__data {
        display: flex;
        flex-direction: column;
        padding: 1em;
        width: 100%;
        .data {
          &__row {
            display: flex;
            td {
              font-size: 1.2em;
              &:first-child {
                width: 50%;
              }
            }
          }
        }
      }
    }
  }
  &--empty {
    @include center-content;
  }
}
