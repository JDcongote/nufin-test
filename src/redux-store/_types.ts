/**Standard Redux type initialization */
export const FETCHING_CONFERENCES = 'FETCHING_CONFERENCES';
export const FETCH_CONFERENCES = 'FETCH_CONFERENCES';
export const FETCHING_TEAMS = 'FETCHING_TEAMS';
export const FETCHED_TEAMS = 'FETCHED_TEAMS';
export const FILTER_TEAMS = 'FILTER_TEAMS';

export const NETWORK_ERROR = 'NETWORK_ERROR';

//states

export interface iTeamsState {
  teams: Team[];
  filteredTeams: Team[];
  conference?: string;
  fetching: boolean;
}

export interface iConferencesState {
  conferences: Conference[];
  fetching: boolean;
}

export interface iErrorState {
  reason: string;
  message: string;
}

//actions
//conferences
interface FetchingConferencesAction {
  type: typeof FETCHING_CONFERENCES;
}

interface FetchConferencesAction {
  type: typeof FETCH_CONFERENCES;
  payload: iConferencesState;
}

//teams
interface FetchingTeamsAction {
  type: typeof FETCHING_TEAMS;
}
interface FetchedTeamsAction {
  type: typeof FETCHED_TEAMS;
  payload: iTeamsState;
}

interface FilterTeamsAction {
  type: typeof FILTER_TEAMS;
  payload: Team[];
}

//types

export interface Team {
  id: number;
  school: string;
  mascot: string;
  abbreviation: string;
  altNames: string[];
  conference: string;
  division: string;
  color: string;
  altColor: string;
  logos: string[];
}

export interface Conference {
  id: number;
  name: string;
  shortName: string;
  abbreviation: string;
}

//export action types
export type ActionTypes =
  | FetchingConferencesAction
  | FetchConferencesAction
  | FetchingTeamsAction
  | FetchedTeamsAction
  | FilterTeamsAction;

export type Error = {
  type: typeof NETWORK_ERROR;
  payload: iErrorState;
};
