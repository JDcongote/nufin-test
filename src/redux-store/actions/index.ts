import {
  iTeamsState,
  iConferencesState,
  FETCHING_CONFERENCES,
  FETCH_CONFERENCES,
  FETCHING_TEAMS,
  FETCHED_TEAMS,
  ActionTypes
} from '../_types';

export function fetchingTeams(): ActionTypes {
  return {
    type: FETCHING_TEAMS
  };
}

export function fetchedTeams(teams: iTeamsState): ActionTypes {
  return {
    type: FETCHED_TEAMS,
    payload: teams
  };
}

export function fetchingConferences(): ActionTypes {
  return {
    type: FETCHING_CONFERENCES
  };
}

export function fetchedConferences(
  conferences: iConferencesState
): ActionTypes {
  return {
    type: FETCH_CONFERENCES,
    payload: conferences
  };
}
