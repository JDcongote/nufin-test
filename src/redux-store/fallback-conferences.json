[
  {
    "id": 1,
    "name": "Kids League 1",
    "shortName": "K1",
    "abbreviation": "KL1"
  },
  {
    "id": 2,
    "name": "Ultra League 1",
    "shortName": "U1",
    "abbreviation": "UL1"
  },
  {
    "id": 3,
    "name": "Super League 1",
    "shortName": "S1",
    "abbreviation": "SL1"
  },
  {
    "id": 4,
    "name": "Big League 2",
    "shortName": "B2",
    "abbreviation": "BL2"
  },
  {
    "id": 5,
    "name": "Core League 2",
    "shortName": "C2",
    "abbreviation": "CL2"
  },
  {
    "id": 6,
    "name": "Mega League 2",
    "shortName": "M2",
    "abbreviation": "ML2"
  },
  {
    "id": 7,
    "name": "Big League 1",
    "shortName": "B1",
    "abbreviation": "BL1"
  },
  {
    "id": 8,
    "name": "Kids League 2",
    "shortName": "K2",
    "abbreviation": "KL2"
  },
  {
    "id": 9,
    "name": "Ultra League 3",
    "shortName": "U3",
    "abbreviation": "UL3"
  },
  {
    "id": 10,
    "name": "Super League 3",
    "shortName": "S3",
    "abbreviation": "SL3"
  },
  {
    "id": 11,
    "name": "Mega League 3",
    "shortName": "M3",
    "abbreviation": "ML3"
  },
  {
    "id": 12,
    "name": "Core League 3",
    "shortName": "C3",
    "abbreviation": "CL3"
  },
  {
    "id": 13,
    "name": "Ultra League 4",
    "shortName": "U4",
    "abbreviation": "UL4"
  },
  {
    "id": 14,
    "name": "Ultra League 2",
    "shortName": "U2",
    "abbreviation": "UL2"
  },
  {
    "id": 15,
    "name": "Big League 4",
    "shortName": "B4",
    "abbreviation": "BL4"
  },
  {
    "id": 16,
    "name": "Big League 3",
    "shortName": "B3",
    "abbreviation": "BL3"
  },
  {
    "id": 17,
    "name": "Mega League 1",
    "shortName": "M1",
    "abbreviation": "ML1"
  },
  {
    "id": 18,
    "name": "Core League 4",
    "shortName": "C4",
    "abbreviation": "CL4"
  },
  {
    "id": 19,
    "name": "Core League 5",
    "shortName": "C5",
    "abbreviation": "CL5"
  },
  {
    "id": 20,
    "name": "Super League 4",
    "shortName": "S4",
    "abbreviation": "SL4"
  }
]
