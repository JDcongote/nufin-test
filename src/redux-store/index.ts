import { TeamReducer } from './reducers/_team-reducer';
import { createStore, applyMiddleware, combineReducers, compose } from 'redux';
import thunk from 'redux-thunk';
import { ConferenceReducer } from './reducers/_conference-reducer';
import { AppReducer } from './reducers/_app-reducer';

/**
 * Combine main app reducers
 */
const rootReducer = combineReducers({
  teamReducer: TeamReducer,
  conferenceReducer: ConferenceReducer,
  appReducer: AppReducer
});

export type AppState = ReturnType<typeof rootReducer>;

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

/**
 * Configure the default store to use the thunk middleware
 */
export default function configureStore() {
  return createStore(rootReducer, composeEnhancers(applyMiddleware(thunk)));
}
