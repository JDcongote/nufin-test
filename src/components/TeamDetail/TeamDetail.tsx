import React from 'react';
import { connect } from 'react-redux';
import { iTeamDetailState, RosterDetail, Team } from 'redux-store/_types';
import { fetchedTeamDetail } from '../../redux-store/actions/';
import { thunkFetchTeamDetail } from '../../thunks';
import Button from '../Common/Button';
import List, { ListItem } from '../Common/list';
import './TeamDetail.scss';

type TeamDetailProps = {
  team: Team;
  close: () => void;
  teamDetail: iTeamDetailState;
  thunkFetchTeamDetail: any;
  loading: boolean;
};
type TeamDetailState = {
  teamDetail: RosterDetail[];
  teamDetailReducer: iTeamDetailState;
  loading: boolean;
  scrolled: boolean;
};

const positionAbbreviations: any[string] = {
  QB: 'Quarterback',
  RB: 'Running Back',
  FB: 'Fullback',
  WR: 'Wide Receiver',
  TE: 'Tight End',
  OL: 'Offensive Lineman',
  C: 'Center',
  G: 'Guard',
  LG: 'Left Guard',
  RG: 'Right Guard',
  T: 'Tackle',
  LT: 'Left Tackle',
  RT: 'Right Tackle',
  K: 'Kicker',
  KR: 'Kick Returner',
  DL: 'Defensive Lineman',
  DE: 'Defensive End',
  DT: 'Defensive Tackle',
  NT: 'Nose Tackle',
  LB: 'Linebacker',
  ILB: 'Inside Linebacker',
  OLB: 'Outside Linebacker',
  MLB: 'Middle Linebacker',
  DB: 'Defensive Back',
  CB: 'Cornerback',
  FS: 'Free Safety',
  SS: 'Strong Safety',
  S: 'Safety',
  P: 'Punter',
  PR: 'Punt Returner'
};

/**
 * Link store state as props
 * @param state
 */
const mapStateToProps = (state: TeamDetailState) => ({
  teamDetail: state.teamDetailReducer,
  loading: state.teamDetailReducer.fetching
});

class TeamDetail extends React.PureComponent<TeamDetailProps, TeamDetailState> {
  componentDidMount() {
    this.props.thunkFetchTeamDetail(this.props.team.school);
  }

  handleClick() {
    this.props.close();
  }
  createPlayers(): ListItem[] {
    return this.props.teamDetail.teamDetail.map((item: RosterDetail) => ({
      key: item.first_name + item.last_name + item.id.toString(),
      fragment: (
        <div className="team-detail__player-card">
          <aside></aside>
          <table className="player-card__data">
            <tbody>
              <tr className="data__row">
                <td>Player:</td>
                <td>
                  {item.first_name} {item.last_name}
                </td>
              </tr>
              <tr className="data__row">
                <td>Weight:</td>
                <td>{item.weight} pounds</td>
              </tr>
              <tr className="data__row">
                <td>Height:</td>
                <td>{(item.height / 12).toFixed(1)} feet</td>
              </tr>
              <tr className="data__row">
                <td>Position:</td>
                <td>{positionAbbreviations[item.position]}</td>
              </tr>
            </tbody>
          </table>
        </div>
      )
    }));
  }

  render() {
    let button = (
      <Button
        text={'Back'}
        floating={'bottom bottom-left'}
        context={this.props.team}
        onClick={this.handleClick.bind(this)}
      ></Button>
    );
    if (this.props.loading) {
      return <div>LOADING</div>;
    } else if (
      this.props.teamDetail &&
      this.props.teamDetail.teamDetail.length === 0
    ) {
      return (
        <React.Fragment>
          <div className="team-detail--empty">NO DATA</div>
          {button}
        </React.Fragment>
      );
    } else {
      return (
        <div className="team-detail">
          <h2 className="team-detail__title">
            {this.props.team.school} Roster
            <span
              className="title__highlight"
              style={{ backgroundColor: this.props.team.color }}
            ></span>
          </h2>
          <List items={this.createPlayers()}></List>
          {button}
        </div>
      );
    }
  }
}

// connect context provider to State store
export default connect(
  mapStateToProps,
  { fetchedTeamDetail, thunkFetchTeamDetail }
)(TeamDetail);
