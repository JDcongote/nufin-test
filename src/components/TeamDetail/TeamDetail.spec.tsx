import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { Team } from 'redux-store/_types';
import configureStore from '../../redux-store';
import TeamDetail from './TeamDetail';

configure({ adapter: new Adapter() });

describe('TeamDetail', () => {
  it('Basic TeamDetail Test', () => {
    const team: Team = {
      id: 103,
      school: 'Boston College',
      mascot: 'Eagles',
      abbreviation: 'BC',
      altNames: ['BC'],
      conference: 'ACC',
      division: 'Atlantic',
      color: '#88001a',
      altColor: '#a39161',
      logos: [
        'http://a.espncdn.com/i/teamlogos/ncaa/500/103.png',
        'http://a.espncdn.com/i/teamlogos/ncaa/500-dark/103.png'
      ]
    };
    const close = jest.fn();

    const render = renderer.create(
      <Provider store={configureStore()}>
        <TeamDetail team={team} close={close} />
      </Provider>
    );
    expect(render.toJSON()).toMatchSnapshot();
  });
});
