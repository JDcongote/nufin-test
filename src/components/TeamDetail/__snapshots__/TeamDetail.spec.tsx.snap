// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamDetail Basic TeamDetail Test 1`] = `
<div
  className="loader"
>
  <div
    className="circle"
  >
    <div
      className="circle__occluder"
    />
  </div>
  <div
    className="circle circle--reverse"
  >
    <div
      className="circle__occluder circle--reverse__occluder"
    />
  </div>
</div>
`;
